Dockerfile

```diff
@@ -26,83 +16,84 @@ RUN python -m venv /venv
 RUN pip install --upgrade "pip==21.3.1"
 RUN useradd -d /app -M --uid 1000 --shell /usr/sbin/nologin kitsune
 
-RUN apt-get update && \
+RUN apt-get update && apt-get install apt-transport-https && \
+    curl -sL https://deb.nodesource.com/setup_12.x | bash - && \
+    apt-get update && \
     apt-get install -y --no-install-recommends \
     gettext build-essential \
     libxml2-dev libxslt1-dev zlib1g-dev git \
     libjpeg-dev libffi-dev libssl-dev libxslt1.1 \
-    libmariadb3 mariadb-client && \
+    libmariadb3 mariadb-client \
+    optipng nodejs zip && \
     rm -rf /var/lib/apt/lists/*
 
 COPY ./requirements/*.txt /app/requirements/
+RUN pip install --no-cache-dir --require-hashes -r requirements/default.txt
 
-RUN pip install --no-cache-dir --require-hashes -r requirements/default.txt && \
-    pip install --no-cache-dir --require-hashes -r requirements/dev.txt
 
-ARG GIT_SHA=head
-ENV GIT_SHA=${GIT_SHA}
+#####################
+# Development image #
+#####################
+FROM base AS dev
 
+RUN pip install --no-cache-dir --require-hashes -r requirements/dev.txt
 
-################################
-# Developer image
-#
-FROM base AS base-dev
-RUN apt-get update && apt-get install apt-transport-https && \
-    curl -sL https://deb.nodesource.com/setup_12.x | bash -
-RUN apt-get update && apt-get install -y --no-install-recommends optipng nodejs zip && \
-    rm -rf /var/lib/apt/lists/*
 
+#########################
+# Frontend dependencies #
+#########################
+FROM base AS base-frontend
```

Comment:
Ah, right, actually it looks like webpack won't necessary solve our need for a python+node environment because of the need to precompile the nunjucks templates. I'm not sure that necessarily needs to be built with the rest of the javascript though, let me experiment with using a "pure" node image for the rest.

Reply:
So the `nunjucks_precompile` python script calls the node `nunjucks-precompile` command, which means we do need a node+python environment.

We could, with relative ease, port this script to node, but if we're porting it to anything we might as well port it to a webpack loader (so we can simply `import "./some/nunjucks/template.html"` wherever we need it).

So my strong inclination is to leave things like this for now, then after/as we merge webpack revisit this.

Reply:
https://github.com/mozilla/sumo-project/issues/973